cabal.project

```diff
@@ -1 +1,6 @@
 packages: *.cabal
+
+tests: True
+benchmarks: True
```

Comment:
Didn't you just remove `test:True` in a recent commit? What's the story here?

Reply:
Previously we used `cabal` mainly for the CI, and we explicitly passed `--disable-tests --disable-benchmarks` to make the build go faster (as we only wanted to check if the library would build). So I thought it wasn't worth it to enable tests in `cabal.project` to then override it to `False` on every command of the CI. It would result in a different behavior in dependency resolution compared to using `cabal` locally without options.

That's also why I cleaned up `cabal.project` to a bare minimum in my previous PR, so that compile-time options would be in the config file shared between both stack and cabal, with little/no overrides in cabal.project.

But now, because dependency resolution doesn't succeed on its own without options, we need to chose:
- use a specific version of `filepath` to obtain a valid build plan; that's commit 8d1533e4754e9b224026f685ce83e65a12b30ba0
- OR enable tests and benchmarks as it coincidentally makes the dependency resolution succeed with a valid build plan; that's commit 919fc6c2fe1e44ae912efaabdc57090304e9d0f2
- OR use `--allow-newer` on a per-package basis, but that would require a lot of trial and error, and we might not find a valid build plan that works with all versions of GHC at once